scarf/merge.py

```diff
@@ -0,0 +1,560 @@
+        assays: List of assay objects to be merged. For example, [ds1.RNA, ds2.RNA].
+        names: Names of each assay objects in the `assays` parameter.
+        mergedCells:
+        nCells: Number of cells in dataset.
+        featCollection:
+        mergedFeats:
+        nFeats: Number of features in the dataset.
+        featOrder:
+        z: The merged Zarr file.
+        assayGroup:
+    """
+
+    def __init__(
+        self,
+        zarr_path: ZARRLOC,
+        assays: list,
+        names: List[str],
+        merge_assay_name: str,
+        in_workspaces: Union[list[str], None] = None,
+        out_workspace: Union[str, None] = None,
+        chunk_size=(1000, 1000),
+        dtype: Optional[str] = None,
+        overwrite: bool = False,
+        prepend_text: Optional[str] = "orig",
+        reset_cell_filter: bool = True,
+        seed: Optional[int] = 42,
+    ):
+        self.assays = assays
+        self.names = names
+        self.inWorkspaces = in_workspaces
+        self.outWorkspace = out_workspace
+        self.merge_assay_name = merge_assay_name
+        self.chunk_size = chunk_size
+        self.permutes_rows, self.permutes_rows_offset, self.coordinates_permutes = (
+            self.perfrom_randomization_rows(seed)
+        )
+        self.mergedCells: pl.DataFrame = self._merge_cell_table(
+            reset_cell_filter, prepend_text
+        )
+        self.nCells: int = self.mergedCells.shape[0]
+        self.featCollection: List[Dict[str, str]] = self._get_feat_ids(assays)
+        self.mergedFeats = self._merge_order_feats()
+        self.nFeats = self.mergedFeats.shape[0]
+        self.featOrder = self._ref_order_feat_idx()
+        self.cellOrder = self._ref_order_cell_idx()
+        self.z = self._use_existing_zarr(zarr_path, merge_assay_name, overwrite)
+        self._ini_cell_data(overwrite)
+        if dtype is None:
+            if len(set([str(x.rawData.dtype) for x in self.assays])) == 1:
+                dtype = str(self.assays[0].rawData.dtype)
+            else:
+                dtype = "float"
+
+        self.assayGroup = create_zarr_count_assay(
+            z=self.z,
+            assay_name=merge_assay_name,
+            workspace=self.outWorkspace,
+            chunk_size=chunk_size,
+            n_cells=self.nCells,
+            feat_ids=np.array(self.mergedFeats["ids"]),
+            feat_names=np.array(self.mergedFeats["names"]),
+            dtype=dtype,
+        )
+
+    def perfrom_randomization_rows(
+        self, seed: Optional[int] = 42
+    ) -> Tuple[
+        Dict[int, Dict[int, np.ndarray]], Dict[int, Dict[int, np.ndarray]], np.ndarray
+    ]:
+        """
+        Perform randomization of rows in the assays.
+        Args:
+            seed: Seed for randomization
+        Returns:
+        """
+        np.random.seed(seed)
+        rowChunks = np.array([x.rawData.chunksize[0] for x in self.assays])
+        rowCells = np.array([x.rawData.shape[0] for x in self.assays])
```

Comment:
```suggestion
        chunkSize = np.array([x.rawData.chunksize[0] for x in self.assays])
        nCells = np.array([x.rawData.shape[0] for x in self.assays])
```
rowChunks can be read as the number of chunks on this row, when it is the size of the chunk (I think 😅).